src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```
@@ -0,0 +1,51 @@
+package mate.academy.service.impl;
+
+import java.util.ArrayList;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user);
```

According to the task description, `ShoppingCartDao.getByUser()` should return an `Optional<ShoppingCart>`, while this service method should return a `ShoppingCart`. This means you need to handle the `Optional` here. You should get the value from the optional or throw an exception if it's not present.